pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/ForwardIndexConfig.java

```diff
@@ -99,6 +171,8 @@ public int hashCode() {
   }
 
   public static class Builder {
+    @Nullable
+    private CompressionCodec _compressionCodec;
```

Comment:
Since we can always tell `_chunkCompressionType` and `_dictIdCompressionType` from `_compressionCodec`, should we remove the other 2 fields?

Reply:
I didn't want to remove them to keep backward compatibility, but I guess we didn't actually release a stable version with `chunkCompressionType` `dictIdCompressionType` so we may not need to worry about that.

Thoughts?

Reply:
IMO we don't need to keep `Builder` backward compatible. We do want to keep the config backward compatible. We may remove `dictIdCompressionType` from the config since that is not released yet, but should keep `chunkCompressionType`

Reply:
Changed